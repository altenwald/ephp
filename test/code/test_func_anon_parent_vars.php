<?php
$msg = 'hello';

$sample = function() {
    var_dump($msg);
};
$sample();

// Inherit $msg
$sample = function() use ($msg) {
    var_dump($msg);
};
$sample();

// The value is passed as a copy, isn't a reference
$msg = 'world';
$sample();

// Reset message
$msg = 'hello';
print "--$msg--\n";

// Inherit by reference
$sample = function() use (&$msg) {
    var_dump($msg);
};
$sample();

// The value is changed in the parent context
$msg = 'world';
$sample();
print "--$msg--\n";

// The closures also accept normal params
$sample = function($arg) use ($msg) {
    var_dump($arg . ' ' . $msg);
};
$sample("hello");
