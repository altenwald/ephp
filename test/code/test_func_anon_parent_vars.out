
Notice: Undefined variable: msg in {{CWD}}/test/code/test_func_anon_parent_vars.php on line 5
NULL
string(5) "hello"
string(5) "hello"
--hello--
string(5) "hello"
string(5) "world"
--world--
string(11) "hello world"
